### PR TITLE
Prevent "Auto Draft" titles and permalinks

### LINKED ISSUE #3508: Avoid "Auto Draft" titles and permalinks
## Bug Description

The "Auto Draft" placeholder should be removed when saving stories, otherwise we get "Auto Draft" titles and "auto-draft" permalinks.

## Expected Behaviour

1. Go to `wp-admin/post-new.php?post_type=web-story` to create a new story
1. Publish the story without making any changes
1. The story's title should be **empty** and the slug is just the story's ID

## Steps to Reproduce

1. Go to `wp-admin/post-new.php?post_type=web-story` to create a new story
1. Publish the story without making any changes
1. See that the story's title is "Auto Draft" and the slug is "auto-draft"

## Additional Context

- Plugin Version: `master`
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Change default title to empty string when the auto draft post is created in PHP. 

## Relevant Technical Choices

It made sense to fix this in PHP, as this is a WordPress problem. 

I had to tweak dashboard to prefill the title with `(No title)` to keep inline with how the core list table works. I did not fill the edit mode in dashboard as this didn't make any sense. 

Removed workaround in header. 

## Testing Instructions

---

Fixes #3508

### CODE REVIEW

**Comment by swissspidy** [comment]
@spacedmonkey Just to clarify, is this how Gutenberg does this? I thought they had a check in JS and not something that filters `wp_insert_post_data`.

See https://github.com/WordPress/gutenberg/blob/b67cd9a9ebb621b8711053fcc44f7030c7dfaee4/packages/core-data/src/actions.js#L449-L463 for reference

**Comment by spacedmonkey** [comment]
> @spacedmonkey Just to clarify, is this how Gutenberg does this? I thought they had a check in JS and not something that filters `wp_insert_post_data`.

I believe we spoke about this in sprint planning. This is a WordPress only problem, so I thought I would fix in WordPress. I don't see why we should a WordPress fix in js, as this codebase will not always be linked to WP.

**Comment by swissspidy** [comment]
Ah you're right.

OK, let me rephrase: does it produce the same result as Gutenberg?

1. Go to wp-admin/post-new.php?post_type=web-story to create a new story
1. Publish the story without making any changes
1. View story on frontend
1. What's the story title and URL?

**Comment by spacedmonkey** [comment]
> 4. What's the story title and URL?

It does what it says in the ticket. Empty title and the id in the url.

**Review by swissspidy — APPROVED** [review]
Works great!

A candidate for an E2E test. I could add one next week.

**Review by BrittanyIRL — APPROVED** [review]
Looks good! Pulled and tested code locally.

**Comment by spacedmonkey** [comment]
This one should be QAed. I will add testing details.

**Comment by swissspidy** [comment]
FWIW I basically did QA and it worked just fine 👍

**Comment by swissspidy** [comment]
@spacedmonkey FYI, there's a small merge conflict

**Comment by spacedmonkey** [comment]
> @spacedmonkey FYI, there's a small merge conflict

Fixed.
